xrootd/streammanager/streammanager.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package streammanager // import "go-hep.org/x/hep/xrootd/streammanager"
+
+import (
+	"math"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/protocol"
+)
+
+type ServerResponse struct {
+	Data  []byte
+	Error error
+}
+
+type dataSendChannel chan<- *ServerResponse
+type DataReceiveChannel <-chan *ServerResponse
+
+const streamIDPartSize = math.MaxUint8
+const streamIDPoolSize = streamIDPartSize * streamIDPartSize
+
+// StreamManager manages channels by their ids.
+// Basically, it's a map[StreamID] chan<-*ServerResponse
+// with methods to claim, free and pass data to specific channel by id.
+type StreamManager struct {
+	dataWaiters map[protocol.StreamID]dataSendChannel
+	freeIds     chan protocol.StreamID
+	mutex       sync.Mutex
```

Comment:
the convention in Go for mutexes is to:
- name them just `mu`
- put them above the field(s) they are supposed to guard